pkg/operator/operator.go

```diff
@@ -200,10 +200,33 @@ func (c *vSphereProblemDetectorController) sync(ctx context.Context, syncCtx fac
 	if _, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, updateFuncs...); updateErr != nil {
 		return updateErr
 	}
-
 	return nil
 }
 
+// runSyncChecks runs vsphere checks and return next duration and whether checks were actually ran.
+func (c *vSphereProblemDetectorController) runSyncChecks(ctx context.Context, clusterInfo *util.ClusterInfo) (time.Duration, bool) {
+	var delay time.Duration
+	if !time.Now().After(c.nextCheck) {
+		return delay, false
+	}
+
+	delay, err := c.runChecks(ctx, clusterInfo)
+	if err != nil {
+		c.lastClusterCheckResult.checkError = err
+		klog.Errorf("Failed to run checks: %s", err)
+	} else {
+		c.lastClusterCheckResult.checkError = nil
+	}
+
+	blockUpgrade, blockUpgradeReason := c.checkForDeprecation(clusterInfo)
+	c.lastClusterCheckResult.blockUpgrade = blockUpgrade
+	c.lastClusterCheckResult.blockUpgradeReason = blockUpgradeReason
+	if blockUpgrade {
+		delay = c.backoff.Step()
```

Comment:
Not related to this PR, but I just noticed it here: if `c.runChecks` fails and `blockUpgrade` is true, both `runChecks` and this line bump the backoff. The backoff takes two steps in a single sync. IMO the backoff manipulation should be just in one function, either in `runChecks` or here.

Reply:
good catch. fixed. Also added test for it.